src/editor/VideoViewer.js

```diff
@@ -0,0 +1,243 @@
+        var id = lang.getId();
+        return id === "video";
+    }
+
+    /**
+     * ImageView objects are constructed when an image is opened
+     * @see {@link Pane} for more information about where ImageViews are rendered
+     *
+     * @constructor
+     * @param {!File} file - The image file object to render
+     * @param {!jQuery} container - The container to render the image view in
+     */
+    function VideoView(file, $container) {
+        this.file = file;
+
+        this.$el = $(Mustache.render(VideoViewTemplate, {
+            videoUrl: _getVideoUrl(file),
+            Strings: Strings
+        }));
+
+        console.log("Before timeout");
+        console.log(_getVideoUrl(file));
+        var that = this;
+
+        setTimeout(function(){
+            console.log("After timeout");
+            console.log(_getVideoUrl(file));
+            that.$videoEl.attr("src", _getVideoUrl(that.file));
+        }, 1000);
+
+        $container.append(this.$el);
+
+        this._naturalWidth = 0;
+        this._naturalHeight = 0;
+
+        this.relPath = ProjectManager.makeProjectRelativeIfPossible(this.file.fullPath);
+
+        // this.$imagePath = this.$el.find(".image-path");
+        this.$videoEl = this.$el.find("video");
+        this.$videoData = this.$el.find(".video-data");
+
+        this.$videoEl.on("canplay", _.bind(this._onVideoLoaded, this));
+        this.$videoEl.on("error", _.bind(console.error, console));
+
+        _viewers[file.fullPath] = this;
+    }
+
+    /**
+     * DocumentManger.fileNameChange handler - when an image is renamed, we must
+     * update the view
+     *
+     * @param {jQuery.Event} e - event
+     * @param {!string} oldPath - the name of the file that's changing changing
+     * @param {!string} newPath - the name of the file that's changing changing
+     * @private
+     */
+    VideoView.prototype._onFilenameChange = function (e, oldPath, newPath) {
+        /*
+         * File objects are already updated when the event is triggered
+         * so we just need to see if the file has the same path as our image
+         */
+        if (this.file.fullPath === newPath) {
+            this.relPath = ProjectManager.makeProjectRelativeIfPossible(newPath);
+        }
+    };
+
+    /**
+     * <video>.on("canplay") handler - updates content of the image view
+     *                             initializes computed values
+     *                            installs event handlers
+     * @param {Event} e - event
+     * @private
+     */
+    VideoView.prototype._onVideoLoaded = function (e) {
+        this._naturalWidth = e.target.videoWidth;
+        this._naturalHeight = e.target.videoHeight;
+
+        var extension = FileUtils.getFileExtension(this.file.fullPath);
+        var stringFormat = Strings.IMAGE_DIMENSIONS;
+        var dimensionString = StringUtils.format(stringFormat, this._naturalWidth, this._naturalHeight);
+
+        this.$videoData.html(dimensionString);
```

Comment:
What about duration info?  Width, height, duration, file size--those seem like the key things for video.